### PR TITLE
Added libonnx recipe

### PR DESCRIPTION
**Pull request by AnthonyBarbier**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by AnthonyBarbier** [comment]
Ping @conda-forge/help-c-cpp: this is ready for review.

**Comment by AnthonyBarbier** [comment]
@conda-forge/staged-recipes : this is ready for review. (Sorry not sure which one I was supposed to use)

File: recipes/libonnx/bld.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+set "PYTHON_EXECUTABLE=%PYTHON%"
+set "PYTHON_LIBRARIES=%LIBRARY_LIB%"
+mkdir build
+if errorlevel 1 exit 1
+cd build
+if errorlevel 1 exit 1
+cmake .. -DCMAKE_INSTALL_PREFIX=%PREFIX% -DONNX_USE_PROTOBUF_SHARED_LIBS=ON -DProtobuf_USE_STATIC_LIBS=OFF -DONNX_USE_LITE_PROTO=ON -DCMAKE_BUILD_TYPE=Release
+if errorlevel 1 exit 1
+cmake --build .
+if errorlevel 1 exit 1
```
Comment (ocefpaf) [inline recipes/libonnx/bld.bat:12]: Is there a `ctest` or `make check/test` that we can here and in the `build.sh`?

File: recipes/libonnx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    - ninja
+    - pybind11
+  run:
+    - python
+    - libprotobuf
+
+test:
+  commands:
+    - conda inspect linkages -p $PREFIX {{ name }}  # [unix]
+    - conda inspect objects -p $PREFIX {{ name }}   # [osx]
```
Comment (ocefpaf) [inline recipes/libonnx/meta.yaml:36]: These are default now in conda-build. Can you add a real test? Or a check if the expected files are created?

**Comment by marcelotrevisani** [comment]
I believe this recipe should be part of the ``onnx`` as an output
What do you think @ocefpaf ?

**Comment by marcelotrevisani** [comment]
Could you please open a PR on this feedstock?
https://github.com/conda-forge/onnx-feedstock

Here you can find some information how to write outputs:
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section

**Comment by ocefpaf** [comment]
> I believe this recipe should be part of the `onnx` as an output
> What do you think @ocefpaf ?

Makes sense. Our fuzzy name linter did not catch this one. Thanks @marcelotrevisani!!

**Comment by xhochy** [comment]
@ocefpaf @marcelotrevisani As far as I understand `libonnx` is independent of Python and thus we would build it several time if we add it as an output to the `onnx-feedstock`? I'm happy to do this work but as I currently struggle a bit with the mixed CI dimension in the `xgboost-feedstock`, I wouldn't like to create that situation again for `libonnx`.

**Pull request closed without merging**